opensearch/src/http/mod.rs

```diff
@@ -37,6 +37,7 @@ pub mod headers;
 pub mod request;
 pub mod response;
 pub mod transport;
+pub mod models;
```

Comment:
This should not be inside the `http` module, as it's not specific to the http implementation/transport layer.

I'd maybe lay it out something like:
```
opensearch
|- src
    |- models
        |- mod.rs
```

Because then we can also eventually put namespace specific models like for `cluster` etc under `models`